### PR TITLE
abstraction of remoting-api for port unification server

### PR DESCRIPTION
**Pull request by TrueAbc**
## What is the purpose of the change

abstraction remoting-api for port unification and netty implementation

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/newportunification/AbstractPortUnificationServer.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.apache.dubbo.remoting.ChannelHandler;
+import org.apache.dubbo.remoting.RemotingException;
+import org.apache.dubbo.remoting.transport.AbstractServer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+// 继承了abstract server作为连接管理handler的能力
```
Comment (wangchengming666) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/newportunification/AbstractPortUnificationServer.java:16]: hi, please use English to replace comment

**Review by wangchengming666 — CHANGES_REQUESTED** [review]

**Comment by EarthChen** [comment]
I suggest to use the server `org.apache.dubbo.remoting.api.PortUnificationServer` of triple protocol directly to replace the current abstract server.
Migrate  server implementations of other protocols directly to this server implementation.

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/newportunification/NewWireProtocol.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.apache.dubbo.remoting.api.newportunification;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.extension.ExtensionScope;
+import org.apache.dubbo.common.extension.SPI;
+
+@SPI(scope = ExtensionScope.FRAMEWORK)
+public interface NewWireProtocol {
```
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/newportunification/NewWireProtocol.java:8]: Do not create new ones, call the original abstract interface

File: dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyTransporter.java [hunk 3]
```diff
@@ -32,7 +33,8 @@ public class NettyTransporter implements Transporter {
 
     @Override
     public RemotingServer bind(URL url, ChannelHandler handler) throws RemotingException {
-        return new NettyServer(url, handler);
+//        return new NettyServer(url, handler);
+        return new NettyPortUnificationServer(url, handler);
```
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyTransporter.java:37]: Do not replace the original logic. `NettyPortUnificationServer` should be created with `NettyPortUnification Transporter`

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

**Pull request closed without merging**